modules/codegen/src/main/scala/com/twilio/guardrail/extract/Extractable.scala

```diff
@@ -2,25 +2,20 @@ package com.twilio.guardrail.extract
 
 import scala.util.Try
 
-
 trait Extractable[T] {
   def extract(v: Any): Try[T]
 }
 
 object Extractable {
-  def build[T](f: PartialFunction[Any, T]): Extractable[T] = {
-    new Extractable[T] {
-      def extract(v: Any): Try[T] = Try(f(v))
-    }
-  }
+  def build[T](f: PartialFunction[Any, T]): Extractable[T] = (v: Any) => Try(f(v))
```

Comment:
I honestly have no idea how this works, how is `(v: Any) => Try(f(v))` an instance of `Extractable`?

Reply:
I think it uses SAM inference, something similar was available for a while since Java 8, intellij actually now suggests it (but you are not using it as far as I know)

Reply:
Yeah, looks like it! I saw all the SAM stuff when it first came out, but I didn't look into it very deeply; it seems this is exactly how that works. Very cool!